### PR TITLE
net: lib: nrf_cloud: Remove sensor attach function and event

### PR DESCRIPTION
**Pull request by jayteemo**
Remove the unnecessary function `nrf_cloud_sensor_attach` and
related event `NRF_CLOUD_EVT_SENSOR_ATTACHED`. This function and
event provided no actual functionality.

Signed-off-by: Justin Morton <justin.morton@nordicsemi.no>

### CODE REVIEW

File: subsys/net/lib/nrf_cloud/src/nrf_cloud.c [hunk 1]
```diff
@@ -259,21 +259,6 @@ int nrf_cloud_shadow_update(const struct nrf_cloud_sensor_data *param)
 	return err;
 }
 
-int nrf_cloud_sensor_attach(const struct nrf_cloud_sa_param *param)
```
Comment (plskeggs) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud.c:262]: Have you tried building `samples/nrf9160/lte_ble_gateway`?  It appears to call `nrf_cloud_sensor_attach()` in `main.c`.

Comment (jayteemo) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud.c:262]: I meant to remove that 🤦🏻‍♂️

**Review by simensrostad — APPROVED** [review]

File: subsys/net/lib/nrf_cloud/src/nrf_cloud.c (hunk 1 above)
Comment (jayteemo) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud.c:262]: removed

**Review by plskeggs — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Comment by rlubos** [comment]
@jayteemo There are conflicts in this PR, please resolve.

**Comment by jayteemo** [comment]
> @jayteemo There are conflicts in this PR, please resolve.

@rlubos done

File: doc/nrf/releases/release-notes-latest.rst [hunk 2]
```diff
@@ -39,6 +39,7 @@ nRF9160
     * Added function :c:func:`nrf_cloud_shadow_device_status_update`, which sets the device status in the device's shadow.
     * Added function :c:func:`nrf_cloud_modem_info_json_encode`, which encodes modem information into a cJSON object formatted for use with nRF Cloud.
     * Added function :c:func:`nrf_cloud_service_info_json_encode`, which encodes service information into a cJSON object formatted for use with nRF Cloud.
+    * Removed function nrf_cloud_sensor_attach, the associated structure nrf_cloud_sa_param, and event NRF_CLOUD_EVT_SENSOR_ATTACHED.  These items provided no useful functionality.
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-latest.rst:42]:
```suggestion
    * Removed function ``nrf_cloud_sensor_attach()``, the associated structure ``nrf_cloud_sa_param``, and event ``NRF_CLOUD_EVT_SENSOR_ATTACHED``. These items provided no useful functionality.
```

**Review by umapraseeda — APPROVED** [review]

**Pull request merged**